torch_xla/distributed/fsdp/xla_fully_sharded_data_parallel.py

```diff
@@ -1398,6 +1399,7 @@ def _free_full_params(self,
     for p in full_params:
       if p._has_full_param:
         # free the original full parameter
+        # RuntimeError: one of the variables needed for gradient computation has been modified by an inplace operation: [XLAFloatType [1000, 2048]] is at version 1; expected version 0 instead. Hint: enable anomaly detection to find the operation that failed to compute its gradient, with torch.autograd.set_detect_anomaly(True).
```

Comment:
This is the error when I was trying to do the replacement.

Reply:
The tricky part was that we hoped to free a tensor `t`'s storage (and reconstruct it later via `t.data = xxx`) without bumping its version counter `t._version`.

I think every time we call `.data` it creates a new Tensor that shares storage but not the version counter. But `.copy_` changes the version counter

Reply:
I also tried to set it to None but PyTorch as a type check... The new value must be a Tensor.